### DISCUSSION TITLE
tcpconnect.bt

### DISCUSSION THREAD

**Discussion by faglanatus** [bpftrace/bpftrace#2071] [General]
![image](https://user-images.githubusercontent.com/33163300/142379672-edeb291b-8a50-47e5-ab18-6388973cd979.png)
*********************************************************************************************************

![image](https://user-images.githubusercontent.com/33163300/142379826-79c1fb08-87af-4c32-b8a2-97d5ea443229.png)

please help me to detect where is error .. im  a bigininer in btftrace langage..

**Comment by danobi** [comment]
Can you share the output of `sudo bpftrace --info` please?

**Reply by faglanatus** [reply]
![image](https://user-images.githubusercontent.com/33163300/142774032-3957691c-3004-4f5e-a563-a0f4d3c35b17.png)
sorry for later

**Reply by fbs** [reply]
Can you copy paste, not screenshot the whole output? This ca't be all of it

**Reply by faglanatus** [reply]
sudo bpftrace ./tcpconnect.bt
[sudo] Mot de passe de natus : 
./tcpconnect.bt:2-4: ERROR: Unrecognized probe type: BENING
after upgrade my ubuntu version 20.4 to 21.4

**Reply by fbs** [reply]
which version are you on? Where did you get bpftrace from? Can you use the build from the docker image we provide to test with?

**Reply by faglanatus** [reply]
sudo bpftrace --info
System
  OS: Linux 5.11.0-40-generic #44-Ubuntu SMP Wed Oct 20 16:16:42 UTC 2021
  Arch: x86_64

Build
  version: v0.11.3
  LLVM: 11
  foreach_sym: yes
  unsafe uprobe: no
  bfd: no
  bpf_attach_kfunc: yes
  bcc_usdt_addsem: yes
  bcc bpf_attach_uprobe refcount: yes
  libbpf: yes
  libbpf btf dump: yes
  libbpf btf dump type decl: yes

Kernel helpers
  probe_read: yes
  probe_read_str: yes
  probe_read_user: yes
  probe_read_user_str: yes
  probe_read_kernel: yes
  probe_read_kernel_str: yes
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: yes

Kernel features
  Instruction limit: 1000000
  Loop support: yes
  btf: yes

Map types
  hash: yes
  percpu hash: yes
  array: yes
  percpu array: yes
  stack_trace: yes
  perf_event_array: yes

Probe types
  kprobe: yes
  tracepoint: yes
  perf_event: yes
  kfunc: yes

output od sudo bpftrace --info

**Comment by faglanatus** [comment]
atus@natus-HP-ProBook-450-G4:~/Bureau/memoirMaster2$ sudo bpftrace --info
Kernel helpers
  get_current_cgroup_id: yes
  send_signal: yes
  override_return: yes

Kernel features
  Loop support: yes

it is all

**Comment by faglanatus** [comment]
> tu es sur quelle version ? D'où avez-vous obtenu bpftrace ? Pouvez-vous utiliser la version à partir de l'image docker que nous fournissons pour tester ?
how can i do docker picture please ?

**Reply by faglanatus** [reply]
cat /etc/os-release
NAME="Ubuntu"
VERSION="21.04 (Hirsute Hippo)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 21.04"
VERSION_ID="21.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=hirsute
UBUNTU_CODENAME=hirsute
that is version of ubuntu
